domain/src/test/groovy/life/qbic/usermanagement/repository/UserRepositorySpec.groovy

```diff
@@ -24,6 +24,34 @@ class UserRepositorySpec extends Specification {
         thrown(RuntimeException)
     }
 
+    def "Given a repository that contains a user with a given email, return the user"() {
+        given:
+        UserDataStorage storage = Mock(UserDataStorage.class)
+        def user = createDummyUser()
+        storage.findUsersByEmail(_ as String) >> [user]
+        UserRepository repository = new UserRepository(storage)
+
+        when:
+        def matchingUser = repository.findByEmail("my.example@example.com")
+
+        then:
+        matchingUser.get().getId().equalsIgnoreCase(user.getId())
```

Comment:
I think in this case an additional check makes understanding test failure easier
```suggestion
        then:
        matchingUser.isPresent()
        matchingUser.get().getId().equalsIgnoreCase(user.getId())
```
Otherwise, you would have to look into a NoSuchElementException for empty Optionals.

Reply:
True, but the test checks explicitly for success :) So if the implementation returns an Optional.empty, this test would fail and the implementation would need to be refactored

Reply:
Yes, I tried it and the matching.isPresent not met was easier to follow up on then the NoSuchElementException from the second check. Both do their job. So this is optional 😛